File: 0199 Binary Tree Right Side View.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new LinkedList<>();
        if (root == null) return res;
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        while (!q.isEmpty()) {
            TreeNode last = null;
            int cnt = q.size();
            for (int i = 0; i < cnt; ++i) {
                last = q.poll();
                if (last.left != null) q.offer(last.left);
                if (last.right != null) q.offer(last.right);
            }
            res.add(last.val);
        }
        return res;
    }
}

File: 0199 Binary Tree Right Side View.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

from collections import deque

class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        res = []
        if not root: return res
        
        q = deque([root])
        while q:
            cnt = len(q)
            for _ in range(cnt):
                last = q.popleft()
                if last.left: q.append(last.left)
                if last.right: q.append(last.right)
            res.append(last.val)
        
        return res